### PR TITLE
fix(Archicad): CNX-9031 CNX-9040 Missing elements in Archicad from revit/objects stream and between localized versions of Archicads

### PR DESCRIPTION
**Pull request by jozseflkiss**
## Description & motivation

https://spockle.atlassian.net/browse/CNX-9031
https://spockle.atlassian.net/browse/CNX-9040

## Changes:

1. C# Connector: Improved exception handling around native-to-native conversion from Revit to Archicad.
2. C++ Add-on: Layer attribute search error ignored

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: ConnectorArchicad/ConnectorArchicad/Converters/Converters/BeamConverter.cs [hunk 1]
```diff
@@ -42,17 +42,19 @@ CancellationToken token
...
+                  new Objects.BuiltElements.Archicad.ArchicadBeam
+                  {
+                    id = beam.id,
+                    applicationId = beam.applicationId,
+                    archicadLevel = Archicad.Converters.Utils.ConvertLevel(beam.level),
+                    begC = Utils.ScaleToNative(baseLine.start),
+                    endC = Utils.ScaleToNative(baseLine.end)
+                  }
+                );
+              }
```
Comment (AlanRynne) [inline ConnectorArchicad/ConnectorArchicad/Converters/Converters/BeamConverter.cs:57]: I'm assuming from this that if a `Beam` doesn't have a line (has something else) we will silently fail?
At least something should be added to the report in DUI I guess?

**Review by AlanRynne — CHANGES_REQUESTED** [review]
Flagging this as "Request changes" but I'm not sure if this is within scope, or rather should be another ticket.

It seems like the previous logic (and current one too) will silently fail a conversion if the geometry is not of the expected type.

Does Archicad not use the Report tab in DUI at all? It seems like we should be notifying the user if a beam/floor/etc is skipped for whatever reason

File: ConnectorArchicad/ConnectorArchicad/Converters/Converters/BeamConverter.cs (hunk 1 above)
Comment (jozseflkiss) [reply ConnectorArchicad/ConnectorArchicad/Converters/Converters/BeamConverter.cs:57]: @AlanRynne You are right about reporting. In these cases we silently fail and the report will be "Unknown". However currently the reporting is implemented on C++ side, so I think this should be another ticket (it is not a regression).

Comment (jozseflkiss) [reply ConnectorArchicad/ConnectorArchicad/Converters/Converters/BeamConverter.cs:57]: Ticket added: https://spockle.atlassian.net/browse/CNX-9060

**Pull request merged**